src/guest_config/src/lib.rs

```diff
@@ -1,5 +1,233 @@
+    )]
+    InvalidFileType(String),
+    #[error("Invalid file format provided. Not able to deserialize guest configuration - [{0}]")]
+    JsonError(serde_json::Error),
+    /// Unknown/Undefined CPU feature name
+    #[error("Unknown or undefined CPU feature name")]
+    UndefinedCpuFeatureName,
+    #[error("CPU feature override for [{0}] is not supported")]
+    UnsupportedCpuFeatureOverride(String),
+    /// Opening or reading the file was unsuccessful
+    #[error("Unable to use file specified [{0}]. See error [{1}]")]
+    IOError(String, std::io::Error),
+    /// Binary file operation was unsuccessful
+    #[error("Error handling binary file [{0}]. See error [{1}]")]
+    BinaryFileOperationError(String, bincode::Error),
+}
+
+/// Takes a file referred to by a template path string and
+/// compiles the json types referred to in the template
+/// into binary objects, converting from a
+/// CustomCpuConfigurationApiRequest to CustomCpuConfiguration
+pub fn compile_template_file(
+    template_file_path_string: &str,
+) -> Result<CustomCpuConfiguration, GuestConfigurationError> {
+    compile_template_string(
+        fs::read_to_string(template_file_path_string).expect(
+            format!(
+                "Unable to read template file [{}]",
+                template_file_path_string
+            )
+            .as_str(),
+        ),
+    )
+}
+
+/// Take a JSON string of CustomCpuConfigurationApiRequest and
+/// converts to the JSON-deserialized type, CustomCpuConfiguration
+pub fn compile_template_string(
+    custom_cpu_config_request_string: String,
+) -> Result<CustomCpuConfiguration, GuestConfigurationError> {
+    match CustomCpuConfigurationApiRequest::try_from(custom_cpu_config_request_string) {
+        Ok(request) => return compile_template_request(&request),
+        Err(err) => Err(err),
+    }
+}
+
+/// Take a JSON string of CustomCpuConfigurationApiRequest and
+/// converts to the JSON-deserialized type, CustomCpuConfiguration
+pub fn compile_template_request(
+    cpu_config_request: &CustomCpuConfigurationApiRequest,
+) -> Result<CustomCpuConfiguration, GuestConfigurationError> {
+    // Validate feature overrides
+    match validate_cpu_feature_overrides(Vec::from(
+        cpu_config_request.cpu_feature_overrides.borrow(),
+    )) {
+        Ok(_) => {
+            // "Compile" general features template
+            match compile_general_arch_template(
+                &cpu_config_request.base_arch_features_template_path,
+            ) {
+                Ok(cpuid) => Ok(CustomCpuConfiguration {
+                    base_arch_features_configuration: cpuid,
+                    cpu_feature_overrides: Vec::from(
+                        cpu_config_request.cpu_feature_overrides.borrow(),
+                    ),
+                }),
+                Err(err) => Err(err),
+            }
+
+            // TODO - Validate+Compile special features template
+        }
+        Err(err) => return Err(err),
+    }
```

Comment:
This can be simplified.
```suggestion
    let _ = validate_cpu_feature_overrides(Vec::from(cpu_config_request.cpu_feature_overrides.borrow()))?;
    // "Compile" general features template
    let cpuid = compile_general_arch_template(&cpu_config_request.base_arch_features_template_path)?;
    // TODO - Validate+Compile special features template
    Ok(CustomCpuConfiguration {
        base_arch_features_configuration: cpuid,
        cpu_feature_overrides: Vec::from(cpu_config_request.cpu_feature_overrides.borrow()),
    })
```